Networking/NetworkingTests/Encoder/ProductOrVariationIDCodableTests.swift

```diff
@@ -0,0 +1,101 @@
+import XCTest
+@testable import Networking
+
+final class ProductOrVariationIDCodableTests: XCTestCase {
+
+    func test_product_encodes_and_decodes_correctly() throws {
+        // Given
+        let productID = ProductOrVariationID.product(id: 999)
+
+        // When
+        let data = try JSONEncoder().encode(productID)
+        let decodedProductID = try JSONDecoder().decode(ProductOrVariationID.self, from: data)
+
+        // Then
+        XCTAssertEqual(productID, decodedProductID)
+    }
+
+    func test_variation_encodes_and_decodes_correctly() throws {
+        // Given
+        let variationID = ProductOrVariationID.variation(productID: 888, variationID: 777)
+
+        // When
+        let data = try JSONEncoder().encode(variationID)
+        let decodedVariationID = try JSONDecoder().decode(ProductOrVariationID.self, from: data)
+
+        // Then
+        XCTAssertEqual(variationID, decodedVariationID)
+    }
+
+    func test_decoding_unknown_type_throws_error() throws {
```

Comment:
👋 Paolo, It looks like this unit test is failing in CI. Could you kindly take a look?

Reply:
Thanks, Sharma. Ready for another take!